### PR TITLE
Updating 21.11 with all changes from 21.7 and 21.3 fro wnprcModules

### PR DESCRIPTION
**Pull request by dnicolalde**
#### Rationale
Planning to update to 21.11, migrating all the changes done to 21.3 until Jan 7, 2022

#### Related Pull Requests
* 

#### Changes
*

### CODE REVIEW

File: WNPRC_Purchasing/src/org/labkey/wnprc_purchasing/WNPRC_PurchasingController.java [hunk 1]
```diff
@@ -296,7 +296,7 @@ private void sendRequestChangeEmailNotification(String oldStatus, EmailTemplateF
                     Map<Integer, String> purchasingDirUserIds = getPurchasingDirectorUserIds();
 
                     //get folder admin users
-                    List<User> adminUsers = SecurityManager.getUsersWithPermissions(getContainer(), Collections.singleton(AdminPermission.class));
```
Comment (dnicolalde) [inline WNPRC_Purchasing/src/org/labkey/wnprc_purchasing/WNPRC_PurchasingController.java:299]: we should keep the 21.7 version

File: WNPRC_Purchasing/test/src/org/labkey/test/tests/wnprc_purchasing/WNPRC_PurchasingTest.java [hunk 2]
```diff
@@ -734,7 +734,7 @@ public void testEmailNotificationApprovalWorkflow()
         goToModule("Dumbster");
         mailTable = new EmailRecordTable(this);
         subject = "Purchase request # " + requestId + " status update";
-        checker().verifyEquals("Incorrect To for the emails sent after approval", Arrays.asList(getCurrentUser(), ADMIN_USER), mailTable.getColumnDataAsText("To"));
```
Comment (dnicolalde) [inline WNPRC_Purchasing/test/src/org/labkey/test/tests/wnprc_purchasing/WNPRC_PurchasingTest.java:737]: Not sure which is the right version here

File: WNPRC_Purchasing/test/src/org/labkey/test/tests/wnprc_purchasing/WNPRC_PurchasingTest.java [hunk 3]
```diff
@@ -779,7 +779,7 @@ public void testEmailNotificationRejectWorkflow()
         goToModule("Dumbster");
         EmailRecordTable mailTable = new EmailRecordTable(this);
         String subject = "Purchase request # " + requestId + " status update";
-        checker().verifyEquals("Incorrect To for the emails sent after rejection", Arrays.asList(getCurrentUser(), ADMIN_USER),
```
Comment (dnicolalde) [inline WNPRC_Purchasing/test/src/org/labkey/test/tests/wnprc_purchasing/WNPRC_PurchasingTest.java:782]: same problem here which one is the right version to move forward

File: WNPRC_Purchasing/src/org/labkey/wnprc_purchasing/WNPRC_PurchasingController.java (hunk 1 above)
Comment (dnicolalde) [reply WNPRC_Purchasing/src/org/labkey/wnprc_purchasing/WNPRC_PurchasingController.java:299]: @gottfred  Can you look at this change?

File: docker/labkey/Dockerfile [hunk 4]
```diff
@@ -28,16 +28,16 @@ RUN if [ "$WNPRC_BRANCH" = "master" ] ; then B="" ; else B=",branch:"${WNPRC_BRA
...
-        LABKEY_TEAMCITY_CONFIG="LabKey_183wnprc_2_Premium_Installers" ; \
-        LABKEY_SNAPSHOT="-SNAPSHOT-" ; \
-        LABKEY_DISTRIBUTION="wisc" ; \
-        LABKEY_FILE_NAME="UWisc-bin.tar.gz" ; \
     elif [ "$LABKEY_VERSION" = 21.3 ] ; then \
         LABKEY_TEAMCITY_CONFIG="LabKey_213Release_External_Wnprc_Installers" ; \
         LABKEY_SNAPSHOT="-SNAPSHOT-" ; \
         LABKEY_DISTRIBUTION="wisc" ; \
         LABKEY_FILE_NAME="UWisc.tar.gz" ; \
+    elif [ "$LABKEY_VERSION" = 21.7 ] ; then \
```
Comment (dnicolalde) [inline docker/labkey/Dockerfile:36]: need to update this to 21.11

File: docker/docker-compose.yml [hunk 5]
```diff
@@ -48,7 +49,7 @@ services:
       - network1
 
   labkey:
-    image: wnprcehr/labkeysnapshot:21.3
+    image: wnprcehr/labkeysnapshot:21.7
```
Comment (dnicolalde) [inline docker/docker-compose.yml:52]: need to update this to 21.11

File: docker/development/docker-compose.yml [hunk 6]
```diff
@@ -6,7 +6,7 @@ networks:
 
 services:
   labkeydev:
-    image: wnprcehr/labkeydev:18.3
+    image: wnprcehr/labkeydev:21.3
```
Comment (dnicolalde) [inline docker/development/docker-compose.yml:9]: need to update this to 21.11

**Review by dnicolalde — COMMENTED** [review]
@gottfred Can you look at some of the changes we made for the purchasing system, not sure which is the right approach

File: WNPRC_Purchasing/src/org/labkey/wnprc_purchasing/WNPRC_PurchasingController.java (hunk 1 above)
Comment (gottfred) [reply WNPRC_Purchasing/src/org/labkey/wnprc_purchasing/WNPRC_PurchasingController.java:299]: This was a patch that Josh asked me to apply in this ticket: https://www.labkey.org/WNPRC/support%20tickets/issues-details.view?issueId=44448

So this change is what we want

File: WNPRC_Purchasing/test/src/org/labkey/test/tests/wnprc_purchasing/WNPRC_PurchasingTest.java (hunk 2 above)
Comment (gottfred) [reply WNPRC_Purchasing/test/src/org/labkey/test/tests/wnprc_purchasing/WNPRC_PurchasingTest.java:737]: These changes are correct, they are the changes that were made to fix the automated tests in this commit:

https://github.com/WNPRC-EHR-Services/wnprc-modules/commit/809a8d2adb07e1b656fbf7adfce005e41760fe46

File: WNPRC_Purchasing/test/src/org/labkey/test/tests/wnprc_purchasing/WNPRC_PurchasingTest.java (hunk 3 above)
Comment (gottfred) [reply WNPRC_Purchasing/test/src/org/labkey/test/tests/wnprc_purchasing/WNPRC_PurchasingTest.java:782]: See above comment

**Comment by dnicolalde** [comment]
Updated the version directly into the new branch. Now the docker files should have reference to the new version

**Pull request merged**